src/platform/plugins/shared/workflows_management/public/features/run_workflow/ui/workflow_execute_index_form.tsx

```diff
@@ -0,0 +1,385 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React, { useEffect, useState, useMemo, useCallback } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiLoadingSpinner,
+  EuiSpacer,
+  EuiText,
+  EuiInMemoryTable,
+  EuiCallOut,
+  EuiToken,
+  EuiDescriptionList,
+  EuiPanel,
+} from '@elastic/eui';
+import { DataViewPicker } from '@kbn/unified-search-plugin/public';
+import { buildEsQuery, type Query, type TimeRange } from '@kbn/es-query';
+import type { DataView, DataViewListItem } from '@kbn/data-views-plugin/public';
+import type { AuthenticatedUser } from '@kbn/security-plugin-types-common';
+import type { SecurityServiceStart } from '@kbn/core-security-browser';
+import { take } from 'rxjs';
+import { useKibana } from '../../../hooks/use_kibana';
+
+interface Document {
+  _id: string;
+  _index: string;
+  _source: {
+    '@timestamp': string;
+    agent?: string;
+    user?: string;
+    [key: string]: any;
+  };
+}
```

Comment:
The `Document` should only define the `_source` shape. For the rest, we can use `SearchHit<Document>`.